### PR TITLE
OT123-257, 361, 318 & 319 | Organizations table and endpoint (various tasks)

### PR DESCRIPTION
**Pull request by DarioMJ48**
[OT123-257](https://alkemy-labs.atlassian.net/jira/software/c/projects/OT123/boards/190?modal=detail&selectedIssue=OT123-257&quickFilter=383)
[OT123-361](https://alkemy-labs.atlassian.net/jira/software/c/projects/OT123/boards/190?modal=detail&selectedIssue=OT123-361&quickFilter=383)
[OT123-318](https://alkemy-labs.atlassian.net/browse/OT123-318)
[OT123-319](https://alkemy-labs.atlassian.net/jira/software/c/projects/OT123/boards/190?modal=detail&selectedIssue=OT123-319)

Organizations endpoint (filter public data by ID).
Organizations table (model & migration).
Organizations table (model & migration) update (social media contact fields).
Organizations endpoint modified (social media links in the response).

wasd

Since me and martin molina 1988 had to create the Organization table, we decided to make them the same way (mostly) to avoid 'conflicts' or solve them easily if they occur.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 node_modules
 .env
+request.http
```
Comment (dvelezroman) [inline .gitignore]: ?

File: app.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ require('dotenv').config()
 
 const indexRouter = require('./routes/index');
 const usersRouter = require('./routes/users');
+const organizationsRouter = require('./routes/organizations');
```
Comment (dvelezroman) [inline app.js]: mover a la carpeta correspondiente

File: app.js [hunk 3]
```diff
@@ -24,6 +25,7 @@ app.use(express.static(path.join(__dirname, 'public')));
 
 app.use('/', indexRouter);
 app.use('/users', usersRouter);
+app.use('/organizations', organizationsRouter);
```
Comment (dvelezroman) [inline app.js]: mover a carpeta routes/index.js

File: controllers/organizations.controller.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+const { Organization } = require("../models")
+
+const controller = {}
+
+controller.byOrganizationID = async (req, res) => {
+    const orgID = req.params.id
+    
+    try {
+      Organization.findOne({ where: { id: orgID } }).then((thisOrganization) => {
+        thisOrganization ? res.json({ name: thisOrganization.name, image: thisOrganization.image, phone: thisOrganization.phone, address: thisOrganization.address, welcomeText: thisOrganization.text}) : res.send('THERE IS NO ORGANIZATION WITH THIS ID.')
```
Comment (dvelezroman) [inline controllers/organizations.controller.js]: line too long...

File: models/organization.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  }
+  Organization.init({
+    name: DataTypes.STRING,
+    image: DataTypes.STRING,
+    phone: DataTypes.STRING,
+    address: DataTypes.STRING,
+    welcomeText: DataTypes.TEXT
+  }, {
+    sequelize,
+    modelName: 'Organization',
```
Comment (dvelezroman) [inline models/organization.js]: seguir ejemplo de #2

**Review by dvelezroman — CHANGES_REQUESTED** [review]

File: controllers/organizations.controller.js (hunk 4 above)
Comment (dvelezroman) [inline controllers/organizations.controller.js]: seguir el patron de la response 
`res.status(statusCode).json({ ok: boolean, result: REsultObject, msg: 'String })`

**Review by dvelezroman — CHANGES_REQUESTED** [review]
Falta completar el modelo

**Comment by DarioMJ48** [comment]
> Falta completar el modelo

¿Qué le falta exactamente? Para hacer los cambios me basé en el que martinmolina1988 tenía aprobado: [acá](https://github.com/alkemyTech/OT123-SERVER/tree/OT123-270/create-slider-and-organization-model).

File: models/organization.js (hunk 5 above)
Comment (DarioMJ48) [reply models/organization.js]: ¿Está bien ahora?

**Review by dvelezroman — APPROVED** [review]
Buen trabajo

**Review by facundoveliz — APPROVED** [review]

**Comment by DarioMJ48** [comment]
OT123-257, 361, 318 & 319. Bien.

**Pull request merged**